File: BlazorLearn_WebApp/BlazorLearn_WebApp.Client/Components/L13/L13_Blazor_MemoryStorage.razor.cs
using Microsoft.AspNetCore.Components;

namespace BlazorLearn_WebApp.Client.Components.L13
{
    public partial class L13_Blazor_MemoryStorage : ComponentBase
    {
        private L13_CacheItemForm? _form;

        [Inject]
        public MemoryStorageAccessor? MemoryStorageAccessor { get; set; }

        private void UpdateState(string state)
        {
            if(_form is not null )
            {
                _form.UpdateOperationState(state);
            }
        }

        private void OnSubmit(CacheItemModel cacheItemModel)
        {
            MemoryStorageAccessor!.Set(cacheItemModel.Key!, cacheItemModel.Value);
            UpdateState($"设置内存存储成功！");
        }

        private void OnGet(CacheItemModel cacheItemModel)
        {
            object? value = MemoryStorageAccessor!.Get(cacheItemModel.Key!);
            if(value is not null )
            {
                UpdateState($"从内存存储中获取到 key {cacheItemModel.Key} 的值为 {value}");
            }
            else
            {
                UpdateState($"内存存储中没有 key {cacheItemModel.Key} 的数据");
            }
        }

        private void OnDelete(CacheItemModel cacheItemModel)
        {
            MemoryStorageAccessor!.Delete(cacheItemModel.Key!);
            UpdateState($"从内存存储中删除 key {cacheItemModel.Key} 的数据成功");
        }

        private void OnClear()
        {
            MemoryStorageAccessor?.Clear();
            UpdateState($"内存存储数据清空成功");
        }
    }
}

File: BlazorLearn_WebApp/BlazorLearn_WebApp.Client/Components/L13/L13_Blazor_MemoryStorage.razor.js

async function getDotNetAssembly() {
    var runtime = globalThis.getDotnetRuntime(0);
    var assembly = await runtime.getAssemblyExports("BlazorLearn_WebApp.Client.dll");
    return assembly;
}

function getKey() {
    return document.getElementById('tb_Key').value;
}

function getValue() {
    return document.getElementById('tb_Val').value;
}

function setStatus(status) {
    document.getElementById('lb_Status').innerHTML = status;
}

async function onSubmit() {
    let asm = await getDotNetAssembly();
    asm.BlazorLearn_WebApp.Client.Components.L13.MemoryStorageAccessorForJs.Set(getKey(), getValue());
    setStatus('Successfully saved data to MemoryStorage from JavaScript!')
}

async function onGet() {
    let asm = await getDotNetAssembly();
    let value = asm.BlazorLearn_WebApp.Client.Components.L13.MemoryStorageAccessorForJs.Get(getKey())
    setStatus('The result of getting ' + getKey() + ' from MemoryStorage in JavaScript is ' + value)
}

async function onDel() {
    let asm = await getDotNetAssembly();
    asm.BlazorLearn_WebApp.Client.Components.L13.MemoryStorageAccessorForJs.Delete(getKey())
    setStatus('Successfully deleted ' + getKey() + ' from MemoryStorage in JavaScript')
}

async function onClear() {
    let asm = await getDotNetAssembly();
    asm.BlazorLearn_WebApp.Client.Components.L13.MemoryStorageAccessorForJs.Clear();
    setStatus('Successfully cleared MemoryStorage from JavaScript')
}